extensions/amp-story/0.1/related-articles.js

```diff
@@ -56,11 +59,14 @@ function buildArticleFromJson_(articleJson) {
 
   user().assert(isProtocolValid(articleJson['url']),
       `Unsupported protocol for article URL ${articleJson['url']}`);
+  dev().assert(articleJson['url']);
 
+  const url = Services.urlForDoc(element);
+  const {host} = url.parse(url.getSourceOrigin(location));
   const article = {
     title: dev().assert(articleJson['title']),
-    url: dev().assert(articleJson['url']),
-    domainName: parseUrlDeprecated(dev().assert(articleJson['url'])).hostname,
+    url: articleJson['url'],
+    domainName: host,
```

Comment:
I may be completely missing the point here, but it looks like we're replacing the domain of the linked article with the current domain the user is on.

Should we / do we want to support the case where a publisher is linking to a link on a different domain?
ie: I'm `gmajoulet.com` and my bookend links to `google.com`, we should display `google.com` as the domain

Reply:
Yeah, I think we want to use `getSourceOrigin` from the url service.  It understands the cache url structure so we don't need to do anything complicated in amp-story.

Reply:
Changed it to use getSourceOrigin on the given url. If the url is relative, it should still take the publishers domain name. 

A caveat here is that I wasn't able to test it locally because `getSourceOrigin` doesn't work with the local server url structure (`http://localhost:8000/proxy/www.google.com...`). But I made sure to catch the error and replace with what we had before in case it fails.